backend/src/main/java/ai/giskard/security/ee/NoAuthConfigurer.java

```diff
@@ -0,0 +1,24 @@
+package ai.giskard.security.ee;
+
+import ai.giskard.domain.User;
+import ai.giskard.repository.UserRepository;
+import org.springframework.security.config.annotation.SecurityConfigurerAdapter;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.web.DefaultSecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+public class NoAuthConfigurer extends SecurityConfigurerAdapter<DefaultSecurityFilterChain, HttpSecurity> {
+
+    private final UserRepository userRepository;
+
+    public NoAuthConfigurer(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public void configure(HttpSecurity http) {
+        User admin = this.userRepository.getOneByLogin("admin");
```

Comment:
The problem is that `NoAuthConfigurer` needs to be created in order to start the backend, but `InitService` that creates default users (including `admin`) is executed once the app starts. So it's a chicken and egg problem that you can reproduce if you remove giskard-home and start from scratch. 
We shouldn't call `userRepository` in the constructor, rather at the incoming request time (`doFilter`)

Reply:
As we discussed => Try to use the login instead of the ID in most cases. This removes the issue altogether